cmd/assetsvc/handler.go

```diff
@@ -120,8 +121,20 @@ func getPaginatedChartList(namespace, repo string, pageNumber, pageSize int, sho
 
 // listCharts returns a list of charts based on filter params
 func listCharts(w http.ResponseWriter, req *http.Request, params Params) {
+
 	pageNumber, pageSize := getPageNumberAndSize(req)
-	cl, meta, err := getPaginatedChartList(params["namespace"], params["repo"], pageNumber, pageSize, showDuplicates(req))
+	namespace, err := url.PathUnescape(params["namespace"])
+	if err != nil {
+		handleDecodeError(params["namespace"], w, err)
+		return
+	}
+	repo, err := url.PathUnescape(params["repo"])
+	if err != nil {
+		handleDecodeError(params["repo"], w, err)
+		return
+	}
```

Comment:
You're right, there's no real need to unescape parms. 
However, note the former nginx behavior prior to this PR was to unescaped everything before the proxy_pass. So all the functions in handler.go do expect something unescaped.
In order to preserve the same behavior and to prevent unnoticed side effect, I opted for manually decoding each param before using it.